### PR TITLE
saferpay api update 1.20 => 1.23

### LINKED ISSUE #45: Implement changes of Saferpay API 1.21
http://saferpay.github.io/jsonapi/#v1.21.0.0.20210323

### PR DESCRIPTION
**Pull request by giriroub**
Closes #45, #46, #53 

This PR contains all changes from 1.20 to 1.23, according to http://saferpay.github.io/jsonapi/#changelog

### CODE REVIEW

**Comment by sprain** [comment]
Thank you very much!
Will try to look into this asap.

File: lib/SaferpayJson/Response/Container/ThreeDs.php [hunk 1]
```diff
@@ -32,10 +32,17 @@ final class ThreeDs
 
     /**
      * @var string|null
-     * @SerializedName("VerificationValue")
+     * @SerializedName("Version")
      * @Type("string")
      */
-    private $verificationValue;
```
Comment (sprain) [inline lib/SaferpayJson/Response/Container/ThreeDs.php:38]: Note to myself: This is a BC break and will therefore result in a new major version.

File: lib/SaferpayJson/Request/Container/SchemeToken.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+     */
+    private $authValue;
+
+    /**
+     * @var string|null
+     * @SerializedName("Eci")
+     */
+    private $eci;
+
+    public function __construct(string $number, int $expMonth, int $expYear, string $authValue)
```
Comment (sprain) [inline lib/SaferpayJson/Request/Container/SchemeToken.php:46]: `$authValue` should be optional, if I see correctly, and therefore not part of the constructor.

Bildschirmfoto 2024-06-04 um 08 38 57

Comment (giriroub) [reply lib/SaferpayJson/Request/Container/SchemeToken.php:46]: It is mandatory in http://saferpay.github.io/jsonapi/1.23/index.html#Payment_v1_Transaction_Initialize , seems they changed it later, but I have not found anything in later release notes. Do you want me to make it optional?

ScreenShot-000886

Comment (sprain) [reply lib/SaferpayJson/Request/Container/SchemeToken.php:46]: Ah great, it's fine therefore for v1.23 👍

**Pull request merged**